test/e2e/apimachinery/resource_quota.go

```diff
@@ -981,6 +988,87 @@ var _ = SIGDescribe("ResourceQuota", func() {
+			},
+			Spec: v1.ResourceQuotaSpec{
+				Hard: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("500m"),
+					v1.ResourceMemory: resource.MustParse("500Mi"),
+				},
+			},
+		}
+		_, err := createResourceQuota(f.ClientSet, ns, resourceQuota)
+		framework.ExpectNoError(err)
+
+		initialResourceQuota, err := f.ClientSet.CoreV1().ResourceQuotas(ns).Get(context.TODO(), rqName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(*initialResourceQuota.Spec.Hard.Cpu(), resource.MustParse("500m"), "Hard cpu value for ResourceQuota %q is %s not 500m.", initialResourceQuota.ObjectMeta.Name, initialResourceQuota.Spec.Hard.Cpu().String())
+		framework.Logf("Resource quota %q reports spec: hard cpu limit of %s", rqName, initialResourceQuota.Spec.Hard.Cpu())
+		framework.ExpectEqual(*initialResourceQuota.Spec.Hard.Memory(), resource.MustParse("500Mi"), "Hard memory value for ResourceQuota %q is %s not 500Mi.", initialResourceQuota.ObjectMeta.Name, initialResourceQuota.Spec.Hard.Cpu().String())
+		framework.Logf("Resource quota %q reports spec: hard memory limit of %s", rqName, initialResourceQuota.Spec.Hard.Memory())
+
+		ginkgo.By(fmt.Sprintf("Updating %q /status", rqName))
+		var updatedResourceQuota *v1.ResourceQuota
+		hardLimits := quota.Add(v1.ResourceList{}, initialResourceQuota.Spec.Hard)
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			updateStatus, err := rqClient.Get(context.TODO(), rqName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Unable to get ResourceQuota %q", rqName)
+			updateStatus.Status = v1.ResourceQuotaStatus{
+				Hard: hardLimits,
+			}
+			updatedResourceQuota, err = rqClient.UpdateStatus(context.TODO(), updateStatus, metav1.UpdateOptions{})
+			return err
+		})
+		framework.ExpectNoError(err, "Failed to update resourceQuota")
+
+		err = wait.PollImmediate(1*time.Second, 30*time.Second, checkResourceQuotaStatus(f, *updatedResourceQuota, hardLimits))
+		framework.ExpectNoError(err, "failed to locate the required values ResourceQuota status")
+
+		ginkgo.By("Patching hard spec values for cpu & memory")
+		xResourceQuota, err := rqClient.Patch(context.TODO(), updatedResourceQuota.Name, types.StrategicMergePatchType,
+			[]byte(`{"spec":{"hard":{"cpu":"1","memory":"1Gi"}}}`),
+			metav1.PatchOptions{})
+		framework.ExpectNoError(err, "Could not patch resourcequota %q. Error: %v", xResourceQuota.Name, err)
+
+		ginkgo.By(fmt.Sprintf("Patching %q /status", rqName))
+		hardLimits = quota.Add(v1.ResourceList{}, xResourceQuota.Spec.Hard)
+
+		rqStatusJSON, err := json.Marshal(hardLimits)
+		framework.ExpectNoError(err)
+		patchedResourceQuota, err := rqClient.Patch(context.TODO(), rqName, types.StrategicMergePatchType,
+			[]byte(`{"status":`+string(rqStatusJSON)+`}`),
+			metav1.PatchOptions{}, "status")
+		framework.ExpectNoError(err)
+
+		err = wait.PollImmediate(1*time.Second, 30*time.Second, checkResourceQuotaStatus(f, *patchedResourceQuota, hardLimits))
```

Comment:
A few questions

1. When you do this check, how do you know that that this e2e test made the change as opposed to the controller?  
2. Does the distinction matter to you?  
3. If you need to be sure that the /status endpoint works, do you do this for other object types by connecting a watch to the resource and looking for the value you updated?
4. Are you verifying that /status properly prevents updates to spec somewhere?

Reply:
I've used watches for other e2e tests for `/status` endpoints in the past but found that they could flake due to missing a watch event (you're not guaranteed to see every watch event). To work around that issue I've had a test not fail on missing a watch event. Switching to `wait.PollImmediate` was a way to have a more reliable test with better guarantees.

I'd not considered the scenario in question 4 as I my focus has been from the list of untested endpoints.